Code/GraphMol/ChemTransforms/MolFragmenter.cpp

```diff
@@ -957,7 +957,7 @@ std::unique_ptr<ROMol> molzip(const ROMol &a, const ROMol &b,
   }
   newmol->updatePropertyCache();
   newmol->setProp(common_properties::_StereochemDone, true);
-  return std::unique_ptr<ROMol>(newmol);
```

Comment:
I'm curious what was leaking here, I had assumed that this was ok to do...

Reply:
ah, it could leak if an exception was thrown somewhere else...

Reply:
I think there was a different problem here: originally `combineMols()` returned a unique_ptr, but the result was being cast to an `RWMol *` up above on line 852. Since the unique_ptr wasn't being assigned to anything, it would immediately be delete'd, so the resulting raw pointer was pointed at memory which was no longer valid.